Fix publishing self-contained ASP.NET apps

dsplaisted opened the pull request:
- Ignore .json files in runtime packs.  Fixes self-contained ASP.NET apps failing to launch with new runtime packs. @pakrym 
- Remove UseRefTargetingPacks property.  Fixes #2962

dsplaisted committed:
Remove UseRefTargetingPacks property

Fixes #2962

dsplaisted committed:
Ignore .json files in runtime packs

pakrym commented on src/Tasks/Microsoft.NET.Build.Tasks/ResolveRuntimePackAssets.cs:
```diff
@@ -71,6 +71,7 @@ void AddAsset(string assetPath, string assetType)
                         assetPath.EndsWith(".map", StringComparison.OrdinalIgnoreCase) ||
                         assetPath.EndsWith(".txt", StringComparison.OrdinalIgnoreCase) ||
                         assetPath.EndsWith(".xml", StringComparison.OrdinalIgnoreCase) ||
+                        assetPath.EndsWith(".json", StringComparison.OrdinalIgnoreCase) ||
```
Should this be an include list instead of exclude list?

dsplaisted replied:
The reason it's not an include list is because native assets can have a variety of extensions and I'm not sure if we know what that list is.

Eventually we would like the runtime packs to [include a manifest](https://github.com/dotnet/cli/issues/10442#issuecomment-454587259) which lists the managed and native files in the pack explicitly, so we don't have to glob for them at all.

nguerrera replied:
https://github.com/dotnet/cli/issues/10537 tracks the manifests.

nguerrera replied:
> native assets can have a variety of extensions and I'm not sure if we know what that list is.

Can't we look at the fixed set of runtime packs we have now? Is it not .dll,.exe,.so,.dylib?

Switching to include list until we have the manifests seems reasonable to me to err on the safer side.

nguerrera replied:
But I'm ok with this more targeted fix to unblock things with the least change given that we have an even safer medium term plan.

dsplaisted committed:
Update 2.2.x shared framework version for tests

dsplaisted committed:
Update stage 0

This version should have a version of the x86 ASP.NET Core runtime pack
which is published to the feeds.

dsplaisted committed:
Add metadata to AppHost items resolved from packs folder

dsplaisted committed:
Add System.Runtime.CompilerServices.Unsafe.dll to expected publish output

dsplaisted commented:
FYI @nguerrera @dotnet/dotnet-cli I had to make a few small fixes to get everything to pass: https://github.com/dotnet/sdk/pull/3002/commits/b81ebc66fdf10cf15fa62e9c333004ba7abf5ed0 and https://github.com/dotnet/sdk/pull/3002/commits/663bab6e32043d66c80482991042c06cf5400076.  I'm going to go ahead and merge.
